core/sqf/src/seatrans/tm/hbasetmlib2/src/main/java/org/trafodion/dtm/HBaseTxClient.java

```diff
@@ -597,24 +602,25 @@ public short doCommit(long transactionId) throws Exception {
        return TransReturnCode.RET_OK.getShort();
    }
 
-   public short completeRequest(long transactionId) throws Exception {
+   public short completeRequest(long transactionId) throws IOException, CommitUnsuccessfulException {
      if (LOG.isDebugEnabled()) LOG.debug("Enter completeRequest, txid: " + transactionId);
      TransactionState ts = mapTransactionStates.get(transactionId);
 
      if(ts == null) {
           LOG.error("Returning from HBaseTxClient:completeRequest, (null tx) retval: " + TransReturnCode.RET_NOTX.toString() + " txid: " + transactionId);
           return TransReturnCode.RET_NOTX.getShort();
        }
-
-       try {
-
-       if (LOG.isTraceEnabled()) LOG.trace("TEMP completeRequest Calling CompleteRequest() Txid :" + transactionId);
-
-          ts.completeRequest();
-       } catch(Exception e) {
-          LOG.error("Returning from HBaseTxClient:completeRequest, ts.completeRequest: txid: " + transactionId + ", EXCEPTION: " + e);
-       throw new Exception("Exception during completeRequest, unable to commit.  Exception: " + e);
-       }
+  
+       boolean loopBack = false;
+       do {
+          try {
+             if (LOG.isTraceEnabled()) LOG.trace("TEMP completeRequest Calling CompleteRequest() Txid :" + transactionId);
+             ts.completeRequest();
```

Comment:
Should there be a loopBack = false after completeRequest in the case where an exception occurred but then the operation succeeded?